### PR TITLE
vimPlugins.nvim-treesitter: fix update.py and update documentation

### LINKED ISSUE #300471: nvim-treesitter update.py fails if ran directly
### Describe the bug
i was unable to run the update.py script inside of pkgs/applications/editors/vim/plugins/nvim-treesitter

### Steps To Reproduce
1. ./pkgs/applications/editors/vim/plugins/nvim-treesitter/update.py

### Expected behavior
the script should run without errors

### Additional context
i wanted to update the nvim-treesitter grammars as some grammars like for instance norg-meta are packaged but aren't included in nvim-treesitter.withAllGrammars

### Notify maintainers
@figsoda @jonringer @softinio @yu-re-ka 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.22, NixOS, 24.05 (Uakari), 24.05.20240327.2726f12`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/y45vqv6pa8bhgag1dw86rvi6rk55xhxn-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by MangoIV**
## Description of changes

fixes https://github.com/NixOS/nixpkgs/issues/300471

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MangoIV** [comment]
cc @teto

**Comment by h7x4** [comment]
You have a small typo in the commit message/PR title, vimPlug~s~ins

**Comment by MangoIV** [comment]
I think I’m incapable of typing the proper thing. They always become vimPlug sins.. maybe a sign? 

Thanks for noticing ;)

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 321535` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

File: doc/languages-frameworks/vim.section.md [hunk 1]
```diff
@@ -214,7 +214,7 @@ Note: this is not possible anymore for Neovim.
...
 
-After running the updater, if nvim-treesitter received an update, also run [`nvim-treesitter/update.py`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/update.py) to update the tree sitter grammars for `nvim-treesitter`.
+After running the updater, if nvim-treesitter received an update, also run [`nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/update.py) to update the tree sitter grammars for `nvim-treesitter`.
```
Comment (teto) [inline doc/languages-frameworks/vim.section.md:217]: ty for the fix. If people are going to run the command like you did, let's put the full path anyway. I've rephrased the beginning hoping it is clearer that one shouldn't have to run the treesitter updater but feel free to ignore:
`When the vim updater detects an nvim-treesitter update, it also runs [pkgs/applications/editors/vim/plugins/nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`] ...`

Comment (MangoIV) [reply doc/languages-frameworks/vim.section.md:217]: Is this what you mean?
```suggestion
When the vim updater detects an nvim-treesitter update, it also runs [`nvim-treesitter/update.py $(nix-build -A vimPlugins.nvim-treesitter)`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vim/plugins/update.py) to update the tree sitter grammars for `nvim-treesitter`.
```

Comment (teto) [reply doc/languages-frameworks/vim.section.md:217]: perfect

**Review by Gerg-L — APPROVED** [review]

**Comment by Gerg-L** [comment]
Second commit could get squash merged in or rebased out

**Pull request merged**